### DISCUSSION TITLE
Hasura | Graphql | Run metadata reload on running server

### DISCUSSION THREAD

**Discussion by rrai10** [hasura/graphql-engine#8999] [General] [answered]
Currently I have no way of reloading the metadata without restarting the server by pushing a new version.  I want to reload the metadata  and my application is on aws server which cant be accessed directly due to api gateway. Hence, I don't have access of Hasura cli on the aws server.
I am using below docker image to run migration and metadata. This only loads on server start. I want to do this while server is running so that when I add enums etc in my table I will get the new metadata while server is running. All in all, just want to know how to run hasura metadata reload on aws with no direct access to backend server .

---docker file
FROM hasura/graphql-engine:v2.8.4.cli-migrations-v3

COPY metadata ./hasura-metadata
COPY migrations ./hasura-migrations

**Comment by BenoitRanque** [comment, marked as answer]
The hasura cli's `hasura metadata reload` relies on the [`reload metadata` api](https://hasura.io/docs/latest/api-reference/metadata-api/manage-metadata/#metadata-reload-metadata) internally. You could use this to trigger a reload without restarting the server.

You will however need access to hasura and the admin secret. There's not really a way to avoid this.
However, you could fully automate this in such a way that hasura reloads itself whenever the conditions for reloading are met.

For example, if you want to reload metadata whenever a record is added to a specific table, you could add an event trigger that calls the reload metadata api. Whether this is the correct path forward depends on your application logic.
